backend/fms_core/import_tool/importers/experiment_run.py

```diff
@@ -0,0 +1,138 @@
+from pandas import pandas as pd
+from fms_core.models import ExperimentType, PropertyType
+from ._generic import GenericImporter
+from fms_core.import_tool.handlers import ExperimentRunHandler
+from .._utils import (data_row_ids_range,
+                      convert_property_value_to_str,
+                      blank_and_nan_to_none)
+
+
+class ExperimentRunImporter(GenericImporter):
+    base_errors = []
+    global_data = {'experiment_type': None, 'protocols_dict': {}, 'property_types_by_name': {}}
+
+    def __init__(self):
+        self.sheet_names = ['Experiments', 'Samples']
+        super().__init__()
+
+    # has to be declared ?? or can directly inherit from its parent?
+    def import_template(self, file, format, dry_run):
+        result = super().import_template(file, format, dry_run)
+        return result
+
+
+    def import_global_data_from_template(self, workflow, properties):
+        # Preload ExperimentType and Protocols dict
+        try:
+            self.global_data['experiment_type'] = ExperimentType.objects.get(workflow=workflow)
+
+            # Preload Protocols objects for this experiment type in a dictionary for faster access
+            self.global_data['protocols_dict'] = self.global_data['experiment_type'].get_protocols_dict
+        except Exception as e:
+            self.base_errors.append(f"No experiment type with workflow {workflow} could be found.")
+
+        # Preload PropertyType objects for this experiment type in a dictionary for faster access
+        for property_column in properties:
+            try:
+                self.global_data['property_types_by_name'][property_column] = PropertyType.objects.get(
+                    name=property_column)
+            except Exception as e:
+                self.base_errors.append(f"Property Type {property_column} could not be found")
+
+
+    def import_template_inner(self):
+        """
+            SAMPLES SHEET
+        """
+        samples_sheet = self.sheets['Samples']
+        samples_row_header = 2
+        samples_sheet.columns = samples_sheet.values[samples_row_header]
+
+        sample_rows_data = []
+
+        for row_id in data_row_ids_range(samples_row_header + 1, samples_sheet):
+            row_data = samples_sheet.iloc[row_id]
+            sample = {'row_id': row_id,
+                      'experiment_id': blank_and_nan_to_none(row_data['Experiment ID']),
+                      'container_barcode': blank_and_nan_to_none(row_data['Source Container Barcode']),
+                      'container_coordinates': blank_and_nan_to_none(row_data['Source Container Position']),
+                      'volume_used': blank_and_nan_to_none(row_data['Source Sample Volume Used']),
+                      'experiment_container_coordinates': blank_and_nan_to_none(row_data['Experiment Container Position'])}
+            sample_rows_data.append(sample)
+
+
+        """
+            EXPERIMENTS SHEET
```

Comment:
Don't fear to break it down in functions / methods Experiment and sample sheets could have their own little function.

Reply:
It is not very modular; which is why i didnt divide it into functions/methods. The only reason why it would be divided would be for better readability. I think those multiple line comments (with more information to be added) will be enough.